packages/api/app/Models/Revision.js

```diff
@@ -0,0 +1,63 @@
+/** @type {typeof import('@adonisjs/lucid/src/Lucid/Model')} */
+const Model = use('Model');
+
+class Revision extends Model {
+	static boot() {
+		super.boot();
+		this.addTrait('Params');
+	}
+
+	reviewer() {
+		return this.belongsTo('App/Models/Reviewer');
+	}
+
+	technology() {
+		return this.belongsTo('App/Models/Technology');
+	}
+
+	attachment() {
+		return this.belongsTo('App/Models/Upload', 'attachment_id');
+	}
+
+	/**
+	 * Runs the revision query with the provided filters.
+	 *
+	 * @param {object} query The query object.
+	 * @param {object} request The request object.
+	 * @returns {object}
+	 */
+	static async scopeWithFilters(query, request) {
+		const filters = request.all();
+
```

Comment:
Melhor mandar o `request.all()` do lado "de fora" pra não dependermos da request aqui no model.
```suggestion
	static async scopeWithFilters(query, filters) {
```

Reply:
@mateus4k, antes era assim, mas em revisões passadas, sugerido por @nicholasio nós decidimos mandar tudo, pois em algum momento pode ser necessário usar o `request.params`, o exemplo disso é o `scopeWithFilters` do model `Technology`:
```javascript
static async scopeWithFilters(query, request) {
		const { params } = request;
		const filters = request.all();
		// we can reuse query scopes from the term model 😎
		if (filters.term) {
			query
				.whereHas('terms', (builder) => {
					builder.getTerm(filters.term);
				})
				.with('terms', (builder) => {
					builder.getTerm(filters.term);
				});
		}

		if (filters.taxonomy) {
			query.with('terms', (builder) => {
				builder.withFilters({ taxonomy: filters.taxonomy });
			});
		}

		if (params.embed) { //<--Nesse trecho de código é usado o params
			query.includeTaxonomy();
		}

		if (filters.status) {
			const statusList = filters.status ? filters.status.split(',') : [];
			if (statusList && statusList.length) {
				query.whereIn('status', statusList);
			}
		}
	}
```